test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATcheckErrorCode to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATcheckErrorCode_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char pString[] = CELLULAR_SAMPLE_SUCCESS_CODE_STRING;
+    char pStringError[] = CELLULAR_SAMPLE_ERROR_CODE_STRING;
+    char * ppKeyList[] = {"TEST1:SUCCESS", "TEST2:ERROR"};
+    size_t keyListLen = 2;
+    bool Result;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATcheckErrorCode( pString, ppKeyList, keyListLen, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    TEST_ASSERT_EQUAL( true, Result );
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATcheckErrorCode( pStringError, ppKeyList, keyListLen, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    TEST_ASSERT_EQUAL( true, Result );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATcheckErrorCode to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATcheckErrorCode_Error_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = "";
+    const char * const * const ppKeyList = {"TEST1:SUCCESS", "TEST2:ERROR"};
+    size_t keyListLen = 2;
+    bool Result;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATcheckErrorCode( pString, ppKeyList, keyListLen, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that any NULL parameter causes Cellular_ATStrStartWith to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATStrStartWith_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString;
+    bool * pResult;
+    char * pPrefix;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATStrStartWith( NULL, NULL, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATStrStartWith( NULL, pPrefix, pResult );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATStrStartWith( pString, NULL, pResult );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATStrStartWith( pString, pPrefix, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+}
+
+/**
+ * @brief Test the error path for Cellular_ATStrStartWith to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATStrStartWith_Error_Path( void )

```

Comment:
test_Cellular_ATStrStartWith_Empty_String
test_Cellular_ATStrStartWith_Empty_Prefix

Reply:
Thanks for the suggestion.